ops/testing.py

```diff
@@ -447,7 +447,7 @@ def begin_with_initial_hooks(self) -> None:
         # the unit status from "Maintenance" to "Unknown". See gh#726
         post_setup_sts = self._backend.status_get()
         if post_setup_sts.get("status") == "maintenance" and not post_setup_sts.get("message"):
-            self._backend.status_set("unknown", "", is_app=False)
+            self._backend.status_set("unknown", "", is_app=False, by_testing_backend=True)
```

Comment:
Instead of adding this new argument, I'd prefer we have have an internal `_status_set()` helper that does it without the check. However, in this case it's trivial, so I think we should just do this:

```suggestion
            self._backend._unit_status = {'status': 'unknown', 'message': ''}
```